python-avd/pyavd/_eos_designs/structured_config/network_services/router_ospf.py

```diff
@@ -76,7 +76,7 @@ def router_ospf(self: AvdStructuredConfigNetworkServices) -> dict | None:
                     if (route_map := get(vrf, "ospf.redistribute_bgp.route_map")) is not None:
                         process_redistribute["bgp"]["route_map"] = route_map
 
-                if get(vrf, "ospf.redistribute_connected.enabled", default=False) is True:
+                if get(vrf, "ospf.redistribute_connected.enabled", default=False) is True and get(vrf, "redistribute_connected", default=True) is True:
```

Comment:
This become confusing as first condition have default as False and second condition have default as True for same config.
SO can change the condition as 
if get(vrf, "redistribute_connected", get(vrf, "ospf.redistribute_connected.enabled", default=False)) is True:

Reply:
Fixed.

Reply:
I don't think we should blend these two knobs. Maybe the new knob should be under BGP and only control that.

Reply:
Fixed this under the comment https://github.com/aristanetworks/avd/pull/4220#discussion_r1692520763